### PR TITLE
Add a Chart for Locust Distributed Load Testing

### PR DESCRIPTION
**Pull request by so0k**
This chart will install Locust master and worker containers.

Please advise on what is required to improve the chart.

Source for container:

- https://github.com/honestbee/distributed-load-testing

### CODE REVIEW

**Comment by so0k** [comment]
@ k8s-ci-robot I signed it!

**Comment by so0k** [comment]
Seems I introduced a problem with the communication from worker to masters when converting to a chart.. if someone could help troubleshoot :(

**Comment by so0k** [comment]
I confirmed the docker images (alpine based) work locally with compose... I'm stumped to find out the cause for my issue :(

https://github.com/honestbee/distributed-load-testing#testing-locust-locally

**Comment by so0k** [comment]
Found & fixed the error, this chart is ready for review

File: incubator/locust/templates/master-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        # livenessProbe:
+        #   periodSeconds: 30
+        #   httpGet:
+        #     path: /
+        #     port: {{ .Values.service.internalPort }}
+        # readinessProbe:
+        #   periodSeconds: 30
+        #   httpGet:
+        #     path: /
+        #     port: {{ .Values.service.internalPort }}
```
Comment (seanknox) [inline incubator/locust/templates/master-deploy.yaml]: Can you uncomment or remove the livenessProbe section?

File: incubator/locust/values.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  externalPort: 8089
+  internalPort: 8089
+  nodePort: 0
+master:
+  config:
+    target-host: https://site.example.com
+  resources:
+    # limits:
+    #   cpu: 100m
+    #   memory: 128Mi
```
Comment (seanknox) [inline incubator/locust/values.yaml]: Uncomment these too.

**Comment by seanknox** [comment]
Hi @so0k, thanks for submitting this. Few changes needed:

1. A chart should be in `stable/` unless it uses alpha APIs (I didn't see any here)
2. Would appreciate a more descriptive `NOTES.txt` for the user. This [drupal example](https://github.com/kubernetes/charts/blob/master/stable/drupal/templates/NOTES.txt) is a good one.
3. Couple minor comments regarding commented out code.

Other than that, this looks LGTM.

**Comment by so0k** [comment]
@seanknox, thanks for the review!

I made following changes according to your comments:

- [x] Move to stable
- [x] Improve NOTES
- [x] uncommented code

While testing my changes on minikube, I also noticed a small bug with the workers starting before the master, so I've added a work around to the `entrypoint` of the locust image and added another env var to the deployment templates.

**Comment by so0k** [comment]
Oops, I accidentally pushed cloudflare datadog chart as well

EDIT: Fixed (rebased and squashed first few commits)

**Comment by so0k** [comment]
Thanks for the fast turnaround

**Pull request merged**